Daniela Lopez Anaya - Astrid Daniela Giraldo Urrea

##**Análisis Casos confirmados por Dengue desde año 2019 hasta el 2021**

La situacion que se va analizar en este caso de estudio es la enfermedad por virus del dengue que se presentó en los últimos tres años en el país de Argentina (2019,2020,2021), a continuacion se presentara el analisis obtenido de las tres bases de datos con su respectivo tratamiento de datos y depuracion.

**Carga de paquetes**

In [2]:
import pandas as pd 
import numpy as np 
import io
import matplotlib as plt
import seaborn as sns
import plotly.express as px
from plotly.subplots import make_subplots
import matplotlib.pyplot as plt
import plotly.io as pio


In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
zika_21 = pd.read_excel('/content/drive/MyDrive/Python/informacion-publica-dengue-zika-nacional-hasta-20210731.xlsx')
dengue20 = pd.read_excel('/content/drive/MyDrive/Python/informacion-publica-dengue-zika-nacional-hasta-20201231_1.xlsx')
dengue19 = pd.read_excel('/content/drive/MyDrive/Python/informacion-publica-dengue-zika-nacional-hasta-20191231_3.xlsx')

##**Base de datos 2021**

Se verifica el tamaño de la base y posibles nulos 

In [5]:
### data frame 2020 
print(zika_21.shape)
print(zika_21.columns)
print(zika_21.isnull().sum())

### dataframe 2021, sus dimensiones son (1146,10)

(1146, 10)
Index(['departamento_id', 'departamento_nombre', 'provincia_id',
       'provincia_nombre', 'año', 'semanas_epidemiologicas', 'evento_nombre',
       'grupo_edad_id', 'grupo_edad_desc', 'cantidad_casos'],
      dtype='object')
departamento_id            0
departamento_nombre        0
provincia_id               0
provincia_nombre           0
año                        0
semanas_epidemiologicas    0
evento_nombre              0
grupo_edad_id              0
grupo_edad_desc            0
cantidad_casos             0
dtype: int64


Se organizo las semanas epidemiologicas en meses

In [6]:
condicion=[(zika_21['semanas_epidemiologicas']<=4),(zika_21['semanas_epidemiologicas'].between(5,9,inclusive=True)),(zika_21['semanas_epidemiologicas'].between(10,13,inclusive=True)),
           (zika_21['semanas_epidemiologicas'].between(14,17,inclusive=True)),(zika_21['semanas_epidemiologicas'].between(18,22,inclusive=True)),(zika_21['semanas_epidemiologicas'].between(23,26,inclusive=True)),
           (zika_21['semanas_epidemiologicas'].between(27,30,inclusive=True)),(zika_21['semanas_epidemiologicas'].between(31,35,inclusive=True)),(zika_21['semanas_epidemiologicas'].between(36,39,inclusive=True)),
           (zika_21['semanas_epidemiologicas'].between(40,44,inclusive=True)),(zika_21['semanas_epidemiologicas'].between(45,48,inclusive=True)),(zika_21['semanas_epidemiologicas'].between(49,53,inclusive=True))]

mes=['enero','febrero','marzo','abril','mayo','junio','julio','agosto','septiembre','octubre','noviembre','diciembre'] 



/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  after removing the cwd from sys.path.


In [7]:
zika_21['mes']=np.select(condicion,mes, default='not specified') ## se crea la columna mes partiendo de las semanas epidemiologicas 

Depuracion y agrupacion devariables

In [8]:
zika_21.grupo_edad_desc = zika_21.grupo_edad_desc.str.lower()
zika_21['grupo_edad_desc']=zika_21['grupo_edad_desc'].replace(['sin especificar', 'edad sin esp.'], 'edad sin esp.') # se agrupan los datos sin especificar
zika_21['departamento_nombre']=zika_21['departamento_nombre'].str.lower()


In [9]:
zika_21.to_csv('/content/drive/MyDrive/Python/zika_final.csv', index=False)

##**Base de dato 2020**

verificar la existencia de nulos en la base



In [10]:

print('Dimensiones: ',dengue20.shape)
print('columnas: ',dengue20.columns)
print('nulos')
print(dengue20.isnull().sum())


Dimensiones:  (11034, 10)
columnas:  Index(['departamento_id', 'departamento_nombre', 'provincia_id',
       'provincia_nombre', 'año', 'semanas_epidemiologicas', 'evento_nombre',
       'grupo_edad_id', 'grupo_edad_desc', 'cantidad_casos'],
      dtype='object')
nulos
departamento_id            0
departamento_nombre        0
provincia_id               0
provincia_nombre           0
año                        0
semanas_epidemiologicas    0
evento_nombre              0
grupo_edad_id              0
grupo_edad_desc            0
cantidad_casos             0
dtype: int64


Se organizan las semanas epidemiologicas en meses

In [11]:
condicion=[(dengue20['semanas_epidemiologicas']<=4),(dengue20['semanas_epidemiologicas'].between(5,9,inclusive=True)),(dengue20['semanas_epidemiologicas'].between(10,13,inclusive=True)),
           (dengue20['semanas_epidemiologicas'].between(14,17,inclusive=True)),(dengue20['semanas_epidemiologicas'].between(18,22,inclusive=True)),(dengue20['semanas_epidemiologicas'].between(23,26,inclusive=True)),
           (dengue20['semanas_epidemiologicas'].between(27,30,inclusive=True)),(dengue20['semanas_epidemiologicas'].between(31,35,inclusive=True)),(dengue20['semanas_epidemiologicas'].between(36,39,inclusive=True)),
           (dengue20['semanas_epidemiologicas'].between(40,44,inclusive=True)),(dengue20['semanas_epidemiologicas'].between(45,48,inclusive=True)),(dengue20['semanas_epidemiologicas'].between(49,53,inclusive=True))]

mes=['enero','febrero','marzo','abril','mayo','junio','julio','agosto','septiembre','octubre','noviembre','diciembre'] 
dengue20['mes']=np.select(condicion,mes, default='not specified') ## se crea la columna mes partiendo de las semanas epidemiologicas

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  after removing the cwd from sys.path.


Depuracion y agrupacion de variables

In [12]:
dengue20.grupo_edad_id = dengue20.grupo_edad_id.str.lower()
dengue20['grupo_edad_id']=dengue20['grupo_edad_id'].replace(['sin especificar', 'edad sin esp.'], 'edad sin esp.') # se agrupan los datos sin especificar


In [13]:
dengue20['departamento_nombre']=dengue20['departamento_nombre'].str.lower()

In [14]:
dengue20['departamento_nombre']=dengue20['departamento_nombre'].replace(['++++++++++++', '*sin dato*'], '*sin dato*') 
dengue20['departamento_nombre']=dengue20['departamento_nombre'].replace(['(en blanco)', '*sin dato*'], '*sin dato*') 
dengue20['departamento_nombre']=dengue20['departamento_nombre'].replace(['#¡valor!', '*sin dato*'], '*sin dato*') 	 # se agrupan los datos sin especificar

In [15]:
dengue20.provincia_nombre.unique()

array(['*sin dato*', 'CABA', 'Formosa', 'Salta', 'Buenos Aires',
       'Catamarca', 'Córdoba', 'Corrientes', 'Chaco', 'Chubut',
       'Entre Ríos', 'Jujuy', 'La Pampa', 'La Rioja', 'Mendoza',
       'Misiones', 'Santa Fe', 'Neuquén', 'Río Negro', 'San Juan',
       'San Luis', 'Santa Cruz', 'Santiago del Estero', 'Tucumán',
       'Tierra del Fuego', '(en blanco)'], dtype=object)

In [16]:
dengue20.provincia_nombre = dengue20.provincia_nombre.str.lower()

In [17]:
dengue20['provincia_nombre']=dengue20['provincia_nombre'].replace(['(en blanco)', '*sin dato*'], '*sin dato*') 

In [18]:
dengue20.año = dengue20.año.apply(lambda x: int(x))

In [19]:
dengue20.to_csv('/content/drive/MyDrive/Python/dengue_2020.csv', index=False)

##**Base de dato 2019**

Verificar existencia de nulos en la base

In [20]:
print('Dimensiones: ',dengue19.shape)
print('columnas: ',dengue19.columns)
print('nulos')
print(dengue19.isnull().sum())

Dimensiones:  (1273, 10)
columnas:  Index(['departamento_id', 'departamento_nombre', 'provincia_id',
       'provincia_nombre', 'año', 'semanas_epidemiologicas', 'evento_nombre',
       'grupo_edad_id', 'grupo_edad_desc', 'cantidad_casos'],
      dtype='object')
nulos
departamento_id            0
departamento_nombre        0
provincia_id               0
provincia_nombre           0
año                        0
semanas_epidemiologicas    0
evento_nombre              0
grupo_edad_id              2
grupo_edad_desc            0
cantidad_casos             0
dtype: int64


como solamente se presentaron dos nulos en la columna grupo_edad_id se procede a rellenarlas con el metodo ffill

In [21]:
dengue19.grupo_edad_id = dengue19.grupo_edad_id.fillna(method="ffill")

Se crea la columna mes partiendo de la columna 'semanas epidemiologicas'

In [22]:
condicion=[(dengue19['semanas_epidemiologicas']<=4),(dengue19['semanas_epidemiologicas'].between(5,9,inclusive=True)),(dengue19['semanas_epidemiologicas'].between(10,13,inclusive=True)),
           (dengue19['semanas_epidemiologicas'].between(14,17,inclusive=True)),(dengue19['semanas_epidemiologicas'].between(18,22,inclusive=True)),(dengue19['semanas_epidemiologicas'].between(23,26,inclusive=True)),
           (dengue19['semanas_epidemiologicas'].between(27,30,inclusive=True)),(dengue19['semanas_epidemiologicas'].between(31,35,inclusive=True)),(dengue19['semanas_epidemiologicas'].between(36,39,inclusive=True)),
           (dengue19['semanas_epidemiologicas'].between(40,44,inclusive=True)),(dengue19['semanas_epidemiologicas'].between(45,48,inclusive=True)),(dengue19['semanas_epidemiologicas'].between(49,53,inclusive=True))]

mes=['enero','febrero','marzo','abril','mayo','junio','julio','agosto','septiembre','octubre','noviembre','diciembre'] 
dengue19['mes']=np.select(condicion,mes, default='not specified') ## se crea la columna mes partiendo de las semanas epidemiologicas 

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated infavour of `both` or `neither`.
  after removing the cwd from sys.path.


Depuracion y agrupacion de variables

In [23]:
dengue19.provincia_nombre = dengue19.provincia_nombre.str.lower()
dengue19.grupo_edad_desc  = dengue19.grupo_edad_desc .str.lower()
dengue19['departamento_nombre']=dengue19['departamento_nombre'].str.lower()

In [24]:
dengue19['departamento_nombre']=dengue19['departamento_nombre'].replace(['(en blanco)', '*sin dato*'], '*sin dato*') 

In [25]:
dengue19['grupo_edad_desc']=dengue19['grupo_edad_desc'].replace(['sin especificar', 'edad sin esp.'], 'edad sin esp.') # se agrupan los datos sin especificar
dengue19.grupo_edad_desc.unique()

array(['de 15 a 19 anos', 'de 25 a 34 anos', 'de 45 a 65 anos',
       'de 35 a 44 anos', 'de 20 a 24 anos', 'mayores de 65 anos',
       'de 10 a 14 anos', 'de 5 a 9 anos', 'de 2 a 4 anos',
       'edad sin esp.', 'posneonato (29 hasta 365 dias)',
       'de 13 a 24 meses'], dtype=object)

In [26]:
dengue19.to_csv('/content/drive/MyDrive/Python/dengue_2019.csv', index=False)

## ***1. ¿En el año 2021 cuales fueron las 10 primeras provincias que se registraron con mayor casos de dengue y en que edades se presentaron los casos?***

In [27]:
zika_21['provincia_nombre']= zika_21['provincia_nombre'].str.lower()
zika_21['grupo_edad_desc'] = zika_21['grupo_edad_desc'].str.lower()
zika_21['grupo_edad_desc']= zika_21['grupo_edad_desc'].replace('de 45 a 64 anos','de 45 a 65 anos')
lista = ['de 5 a 9 anos','de 10 a 14 anos', 'de 15 a 19 anos','de 20 a 24 anos','de 25 a 34 anos','de 35 a 44 anos','de 45 a 65 anos']
e = pd.DataFrame(lista,columns=['grupo_edad_desc'])
zika_edades = zika_21.groupby(['grupo_edad_desc','provincia_nombre'])[['cantidad_casos']].sum().reset_index()

zika_edades = zika_edades.sort_values(by='cantidad_casos', ascending= False)
zika_edades['grupo_edad_desc']=zika_edades['grupo_edad_desc'].astype('category')
zika_edades['provincia_nombre']=zika_edades['provincia_nombre'].astype('category')
zika_ed= zika_edades.head(10).reset_index(drop=True)
zika_edades1 = pd.merge(e, zika_ed, on='grupo_edad_desc', how='outer')
zika_edades1


grupo_edad_desc provincia_nombre  cantidad_casos
0    de 5 a 9 anos            salta             139
1  de 10 a 14 anos            salta             240
2  de 15 a 19 anos            salta             206
3  de 20 a 24 anos            salta             200
4  de 25 a 34 anos            salta             372
5  de 25 a 34 anos         la rioja             231
6  de 35 a 44 anos            salta             299
7  de 35 a 44 anos         la rioja             170
8  de 45 a 65 anos            salta             355
9  de 45 a 65 anos         la rioja             228

In [28]:
fig = px.bar(zika_edades1, x = 'grupo_edad_desc', y='cantidad_casos', color = 'provincia_nombre', barmode = 'group', title= '<b> Mayor numero de casos segun provincia <b>',
             color_discrete_sequence=["darkcyan","coral"]) 

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Grupo edad',
    yaxis_title = 'Cantidad Casos',
    template = 'simple_white',
    title_x = 0.5)

fig.show()


*Realizando un analisis a la grafica anterior, la cual muestra las 10 primeras provincias que  presenta los mayor casos de dengue y con sus respectivo grupo de edades, se concluye que la provincia de salta fue la mas afecta durante este año, debido a se registraron mas casos de dengue y en diferentes grupo de edades, por otro lado podemos observar que el grupo de edad mas afectada es entre 25 y 34 años con un total de 372 casos en la provincia de salta y con un total de 231  en la provincia la rioja.*

##**2. ¿Cual es el porcentaje de  casos de dengue que se presentaron por mes en el año 2021?**

In [29]:
m = zika_21.groupby(['mes'])[['cantidad_casos']].sum().reset_index()
m

mes  cantidad_casos
0      abril            1401
1  diciembre               4
2      enero             112
3    febrero             539
4      junio              78
5      marzo             785
6       mayo             931

In [30]:
fig = px.pie(m, values = 'cantidad_casos', names ='mes',
             title= '<b>Porcentaje de casos segun el mes año 2021<b>',
             color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles a la gráfica:
fig.update_layout(
    xaxis_title = '',
    yaxis_title = '',
    template = 'simple_white',
    title_x = 0.5)

fig.show()

Analizando la grafica anterior se observa que los meses que se presentaron mayores casos de dengue fueron el mes de abril con 36,4 %, el mes de mayo con un 24.2%, el mes de marzo con un 20,4% y el mes de febrero con un 14%, ademas es importante resaltar que el 99% de los casos de dengue se presentaron en el primer semestre del año.

##**3.¿Cual fue el departamento mas afectado por el dengue en el 2020 y en que mes presento mas casos registrados?**

In [31]:
dengue20_departamento = dengue20.groupby(['departamento_nombre'])[['cantidad_casos']].sum().reset_index()
dengue20_departamento = dengue20_departamento.sort_values(by='cantidad_casos', ascending= False)
dengue20_departamento.head(1)

departamento_nombre  cantidad_casos
37             capital         13325.0

In [32]:
dengue20_departamento.tail(1)

departamento_nombre  cantidad_casos
238  san blas de los sauces             1.0

Al realizar el respectivo analisis de los departamentos de argentina afectados, podemos concluir que el departamento con mas registros de Dengue es la capital, con un registro de 13325 casos, asi mismo se puede observar que el departamento con menos registros es el departamento san blas de los sauces con total de un caso registrado.

In [33]:
dengue20_mes_dep = dengue20.groupby([dengue20['departamento_nombre'][dengue20.departamento_nombre == 'capital'], 'mes'])[['cantidad_casos']].sum().reset_index()
dengue20_mes_dep= dengue20_mes_dep.sort_values(by='cantidad_casos', ascending= False)
lista = ['enero','febrero','marzo','abril','mayo','junio','julio','agosto','septiembre','octubre','noviembre','diciembre']
d = pd.DataFrame(lista,columns=['mes'])
dengue20_mes_dep1 = pd.merge(d, dengue20_mes_dep, on='mes', how='outer')
dengue20_mes_dep1.head()

mes departamento_nombre  cantidad_casos
0    enero             capital            69.0
1  febrero             capital           616.0
2    marzo             capital          2889.0
3    abril             capital          5384.0
4     mayo             capital          4143.0

In [34]:
dengue20_mes_dep.tail(3)

departamento_nombre         mes  cantidad_casos
1             capital      agosto             3.0
2             capital   diciembre             2.0
9             capital  septiembre             1.0

De la anterior informacion se puede observar que el mes  mas afectado en la capital es el mes de abril con un cantidad de 5384 casos registrados y el mes menos afectado es el mes de septiembre con el registro solamente de un caso.

In [35]:
#Crear grafica
fig = px.bar(dengue20_mes_dep1, x = 'mes', y='cantidad_casos',
             title= '<b>Total de casos registrados en la capital<b>',
             color_discrete_sequence = px.colors.qualitative.G10)
# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Mes',
    yaxis_title = 'Total casos resgitrados',
    template = 'simple_white',
    title_x = 0.5)
fig.show()


Con la grafica obtenida anteriormnete se puede concluir que el mes mas afectado efectivamente es el mes de abril y los meses menos efectados, teniendo como  registro un solo caso, son los meses de septiembre, agosto y diciembre.

##**4.¿Cual es el porcentaje de casos de dengue por edades en el año 2020?** 

In [36]:
edades_20 = dengue20.groupby(['grupo_edad_id'])[['cantidad_casos']].sum().reset_index()

In [37]:
edades_20 = edades_20.sort_values(by='cantidad_casos', ascending= False)
edades_20.tail(5)

grupo_edad_id  cantidad_casos
10                  edad sin esp.           848.0
1                de 13 a 24 meses           280.0
14  posneonato (de 29 a 365 días)           137.0
13  posneonato (de 29 a 365 dias)           125.0
12        neonato (hasta 28 dias)             1.0

In [38]:
fig = px.pie(edades_20, values = 'cantidad_casos', names ='grupo_edad_id',
             title= '<b>Porcentaje de casos segun la edad año 2020<b>',
             color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles a la gráfica:
fig.update_layout(
    xaxis_title = '',
    yaxis_title = '',
    template = 'simple_white',
    title_x = 0.5)

fig.show()

De acuerdo a la informacion obtenida anteriormente, se puede observar que la edad mas efcatada en el año 2020 fue la edad en el rango de 25 a 34 años, representado el 20,3%, y de la tabla podemos observar que las edades menos efectadas son los de menos de un año de edad.

##**5.¿Cual es la provincia de argentina donde se presentan mas casos y que mes es el mas afectado en el 2019, esto cambia calculando el mes mas afectado en general?**

In [39]:
dengue19_provincia = dengue19.groupby(['provincia_nombre'])[['cantidad_casos']].sum().reset_index()
dengue19_provincia = dengue19_provincia.sort_values(by='cantidad_casos', ascending= False)
dengue19_provincia.head(5)

provincia_nombre  cantidad_casos
7            salta             944
6         misiones             572
5            jujuy             554
8         santa fe             495
2            chaco             120

In [40]:
dengue19_provincia.tail(5)

provincia_nombre  cantidad_casos
1                  caba              24
4               formosa              14
10              tucumán               5
9   santiago del estero               4
3               córdoba               3

In [41]:
#Crear grafica
fig = px.bar(dengue19_provincia, x = 'provincia_nombre', y='cantidad_casos',
             title= '<b>Total de casos registrados por provincia en el 2019<b>',
             color_discrete_sequence=["cadetblue"])
# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Provincia',
    yaxis_title = 'Total casos resgitrados',
    template = 'simple_white',
    title_x = 0.5)
fig.show()


De la informacion obtenida anteriormente se puede observar que la provincia mas afectada por el dengue en argentina en el año 2019, fue la provincia salta con un total de 944 casos registrados, asi mismo se puede observar que la provincia con menos casos registrados, 3 casos, es la provincia de cordoba.

In [42]:
dengue19_mes = dengue19.groupby([dengue19['provincia_nombre'][dengue19.provincia_nombre == 'salta'], 'mes'])[['cantidad_casos']].sum().reset_index()
dengue19_mes= dengue19_mes.sort_values(by='cantidad_casos', ascending= False)
dengue19_mes.head(5)

provincia_nombre      mes  cantidad_casos
7            salta     mayo             268
6            salta    marzo             240
3            salta  febrero             205
0            salta    abril             175
5            salta    junio              24

In [43]:
dengue19_mes.tail(5)

provincia_nombre        mes  cantidad_casos
0            salta      abril             175
5            salta      junio              24
2            salta      enero              22
4            salta      julio               8
1            salta  diciembre               2

De las tablas anteriores se puede observar que la provincia argentina salta fue mas afectada en el mes de mayo y marzo teniendo un total de 308 casos registrados y el mes con menos casos fue el de diciembre con 2 casos registrados.

In [44]:
meses = dengue19.groupby(['mes'])[['cantidad_casos']].sum().reset_index()

In [45]:
fig = px.pie(meses, values = 'cantidad_casos', names ='mes',
             title= '<b>Porcentaje de casos segun el mes año 2019<b>',
             color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles a la gráfica:
fig.update_layout(
    xaxis_title = 'mes',
    yaxis_title = 'casos',
    template = 'simple_white',
    title_x = 0.5)

fig.show()

Con la grafica obtenida anteriormente se puede observar el mes mas efectado tanto para lka provincia como a nivel general fue el mes de mayo y el mes menos efctado si cambio, siendo esta vez el del mes de septiembre con un total de tres casos registrados.

##**6¿Cual es el rango de edad con mas casos en el año 2019 y en que mes fue el mas efectado?**

In [46]:
dengue19_edades = dengue19.groupby(['grupo_edad_desc'])[['cantidad_casos']].sum().reset_index()
dengue19_edades = dengue19_edades.sort_values(by='cantidad_casos', ascending= False)
dengue19_edades.head(5)

grupo_edad_desc  cantidad_casos
7  de 45 a 65 anos             635
5  de 25 a 34 anos             529
6  de 35 a 44 anos             435
2  de 15 a 19 anos             307
4  de 20 a 24 anos             298

In [47]:
fig = px.pie(dengue19_edades, values = 'cantidad_casos', names ='grupo_edad_desc',
             title= '<b>Porcentaje de casos segun el rango de edad año 2019<b>',
             color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles a la gráfica:
fig.update_layout(
    xaxis_title = '',
    yaxis_title = '',
    template = 'simple_white',
    title_x = 0.5)

fig.show()

En la anterior tabla y en el grafico, se puede observar que las perosnas mas afectadas por el dengue en el 2019 fueron personas entre los 45 y 65 años de edad, reprsentado el 22.7% y 635 casos registrados, diferente en comparacion con el año 2020 en donde el rango de edad mas afectado es entre los 25 a 34 años.

In [48]:
dengue19_mes1 = dengue19.groupby([dengue19['grupo_edad_desc'][dengue19.grupo_edad_desc == 'de 45 a 65 anos'], 'mes'])[['cantidad_casos']].sum().reset_index()
dengue19_mes1= dengue19_mes1.sort_values(by='cantidad_casos', ascending= False)
dengue19_mes1.head(5)

grupo_edad_desc      mes  cantidad_casos
7  de 45 a 65 anos     mayo             194
0  de 45 a 65 anos    abril             183
6  de 45 a 65 anos    marzo             132
3  de 45 a 65 anos  febrero              86
2  de 45 a 65 anos    enero              19

In [49]:
dengue19_mes1.tail(5)

grupo_edad_desc         mes  cantidad_casos
5  de 45 a 65 anos       junio              13
1  de 45 a 65 anos   diciembre               3
4  de 45 a 65 anos       julio               2
8  de 45 a 65 anos   noviembre               2
9  de 45 a 65 anos  septiembre               1

se puede observar que el mes donde se registraron mas casos en este rango de edad fue el mes de mayo, al igual que a nivel general, con un total de 194 casos registrados, y el mes menos afectado fue el mes de septiembre con el registro de un caso.

##**7.¿Cual ha sido el comportamiento por meses del dengue en los ultimos tres años?**

In [50]:
dengue19_1 = dengue19.groupby(['mes'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2019'}).reset_index() ## SE CREA DATA AGRUPADA por mes
dengue20_1 = dengue20.groupby(['mes'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2020'}).reset_index() 
zika_21_1 = zika_21.groupby(['mes'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2021'}).reset_index()

In [51]:
lista = ['enero','febrero','marzo','abril','mayo','junio','julio','agosto','septiembre','octubre','noviembre','diciembre']
d = pd.DataFrame(lista,columns=['mes'])
df = pd.merge(d, dengue19_1, on='mes', how='outer')
df = pd.merge(df, dengue20_1, on='mes', how='outer')
df = pd.merge(df,zika_21_1, on = 'mes', how='outer')

In [52]:
df = df.fillna(0)

In [53]:
df['TOTAL'] = df.iloc[:,1:].apply(lambda x: sum(x), axis=1)

In [54]:
fig= px.line(df, x='mes', y=['2019','2020','2021'],
             title='<b>Casos de dengue en Argentina del 2019 al 2021<b>', 
             color_discrete_sequence=px.colors.qualitative.G10)

#agregar detalles
fig.update_layout(
    template='simple_white',
    title_x=0.5,
    legend_title = 'Año',
    xaxis_title= '<b>Meses<b>',
    yaxis_title= '<b>Cantidad de casos por mes <b>'
) 
fig.show()

Se puede observar de la grafica anterior que el año en el que se presento un cambio significativo y alarmante es el año 2020, donde su pico mas alto es el mes de abril con un total de 3115 casos registrados, para los años 2019 y 2021 presentan un comportamiento similar, donde su pico mas alto se observa en el mes de mayo para el 2019 con un total de 363 casos y en abril para el 2021 con un total 336 casos, así mismo se puede observar que en los meses de noviembre, octubre, septiembre y agosto, tienen prácticamente un comportamiento similar en cada año.

##**8 ¿Cual es el rango de edad mas efectado en los ultimos tres años?**

In [55]:
dengue20 = dengue20.rename(columns = { 'grupo_edad_desc': 'grupo_edad'})

In [56]:
dengue20 = dengue20.rename(columns = { 'grupo_edad_id': 'grupo_edad_desc'})

In [57]:
dengue19_2 = dengue19.groupby(['grupo_edad_desc'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2019'}).reset_index() ## SE CREA DATA AGRUPADA por mes
dengue20_2 = dengue20.groupby(['grupo_edad_desc'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2020'}).reset_index() 
zika_21_2 = zika_21.groupby(['grupo_edad_desc'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2021'}).reset_index()

In [58]:
df1 = pd.merge(dengue19_2, dengue20_2, on='grupo_edad_desc', how='outer')
df1 = pd.merge(df1,zika_21_2, on = 'grupo_edad_desc', how='outer')

In [59]:
df1 = df1.fillna(0)
df1['TOTAL'] = df1.iloc[:,1:].apply(lambda x: sum(x), axis=1)
df1= df1.sort_values('TOTAL', ascending=False)
df1['grupo_edad_desc']=df1['grupo_edad_desc'].replace(['de 45 a 65 anos', 'de 45 a 64 anos'], 'de 45 a 64 anos')
df1['grupo_edad_desc']=df1['grupo_edad_desc'].replace(['posneonato (29 hasta 365 dias)', 'posneonato (de 29 a 365 dias)'], 'posneonato (de 29 a 365 dias)')
df1['grupo_edad_desc']=df1['grupo_edad_desc'].replace(['posneonato (de 29 a 365 días)', 'posneonato (de 29 a 365 dias)'], 'posneonato (de 29 a 365 dias)')
lista = ['neonato (hasta 28 dias)','posneonato (de 29 a 365 dias)','de 13 a 24 meses','de 2 a 4 anos','de 5 a 9 anos','de 10 a 14 anos','de 15 a 19 anos','de 20 a 24 anos','de 25 a 34 anos','de 35 a 44 anos','de 45 a 64 anos','mayores de 65 anos']
d = pd.DataFrame(lista,columns=['grupo_edad_desc'])
df1 = pd.merge(d, df1, on='grupo_edad_desc', how='outer')
df1

grupo_edad_desc   2019    2020   2021   TOTAL
0         neonato (hasta 28 dias)    0.0     1.0    1.0     2.0
1   posneonato (de 29 a 365 dias)    0.0   116.0    0.0   116.0
2   posneonato (de 29 a 365 dias)    0.0    42.0    0.0    42.0
3   posneonato (de 29 a 365 dias)    4.0     0.0   13.0    17.0
4                de 13 a 24 meses    4.0   163.0   10.0   177.0
5                   de 2 a 4 anos   40.0   426.0   41.0   507.0
6                   de 5 a 9 anos   82.0   800.0   89.0   971.0
7                 de 10 a 14 anos  127.0   993.0  116.0  1236.0
8                 de 15 a 19 anos  140.0  1194.0  140.0  1474.0
9                 de 20 a 24 anos  151.0  1340.0  133.0  1624.0
10                de 25 a 34 anos  220.0  1729.0  189.0  2138.0
11                de 35 a 44 anos  190.0  1637.0  165.0  1992.0
12                de 45 a 64 anos  226.0  1579.0  164.0  1969.0
13                de 45 a 64 anos    0.0   128.0    0.0   128.0
14             mayores de 65 anos   87.0   845.0   84.0  1016.0
15                  edad sin esp.    2.0    41.0    1.0    44.0

In [60]:
y=sum(df1.TOTAL)
Porcentaje_edad_mas_afectada = 4276.0/y
Porcentaje_edad_mas_afectada 

0.31784732030030477

In [61]:
#Crear grafica
fig = px.bar(df1, x = 'grupo_edad_desc', y='TOTAL',
             title= '<b>Total de casos por grupo de edad<b>',
             color_discrete_sequence=["cadetblue"])
# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Edad',
    yaxis_title = 'Total casos registrados',
    template = 'simple_white',
    title_x = 0.5,
   )

pio.show(fig)


Con los datos obtenidos anteriormente se puede observar que la personas mas afectadas en los ultimos tres años por dengue son las personas entre 25 y 34 años de edad con un total de 4276 casos registrados representadio el 15% del total de los registros, la organizacion mundial de la salud (OMS) notifica que el virus del dengue puede afectar a cualquier persona de cualquier edad, que los casos de dengue grave son más frecuentes en personas que ya padecieron dengue por un serotipo (infección primaria) y se infectan nuevamente (infección secundaria) con un serotipo diferente, pero esto no implica necesariamente que toda infección secundaria conduzca a dengue grave.

##**9¿Cual es el departamento mas afectado por el dengue en los ultimos tres años?**

In [62]:
dengue19_3 = dengue19.groupby(['departamento_nombre'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2019'}).reset_index() ## SE CREA DATA AGRUPADA por mes
dengue20_3 = dengue20.groupby(['departamento_nombre'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2020'}).reset_index() 
zika_21_3 = zika_21.groupby(['departamento_nombre'])[['cantidad_casos']].count().rename(columns = {'cantidad_casos':'2021'}).reset_index()

In [63]:
df2 = pd.merge(dengue19_3, dengue20_3, on='departamento_nombre', how='outer')
df2 = pd.merge(df2,zika_21_3, on = 'departamento_nombre', how='outer')
df2 = df2.fillna(0)
df2['TOTAL'] = df2.iloc[:,1:].apply(lambda x: sum(x), axis=1)
df2= df2.sort_values('TOTAL', ascending=False)
df2 = df2.head(5)
df2

departamento_nombre   2019    2020   2021   TOTAL
8                    capital  105.0  1068.0  230.0  1403.0
176  grl. josé de san martín    0.0   378.0  165.0   543.0
232                     orán    0.0   225.0  226.0   451.0
4                       anta   36.0   277.0  116.0   429.0
42                   ledesma  114.0   165.0   77.0   356.0

**Departamento mas afectado a nivel general**

In [64]:
cnt_ed = df2['TOTAL'].sum()

# hacer la gráfica
fig = px.pie(df2, values = 'TOTAL', names ='departamento_nombre',
             title= '<b>% Departamentos mas afectados por el dengue <b>', hole = .5,
             color_discrete_sequence=px.colors.qualitative.G10)

# poner detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    annotations = [dict(text = str(cnt_ed), x=0.5, y = 0.5, font_size = 40, showarrow = False )])
fig.show()

Gracias a la informacion obtenida anteriormente el departamento mas efectado en argentina en los ultimos años es el departamento denotado como capital obteniendo el 44.1% de los datos, el departamento Capital es el nombre de varios departamentos de provincias argentinas, llamados así pues allí se encuentran las capitales de provincia, por lo cual se procede a realizar un analisi indiviual por año para observar cual es el departamento con mas casos y si es la capital a que provincia se refiere, ya que en la informacion obtenida anteriormente se tiene el total de los casos en el departamento capital sin tener en cuenta a la provincia que pertenece.

**Departamento mas afectado en el año 2019**

In [65]:
df3= df2.sort_values('2019', ascending=False)
df3=df3.head(3)


In [66]:
cnt_ed = df3['2019'].sum()

# hacer la gráfica
fig = px.pie(df3, values = '2019', names ='departamento_nombre',
             title= '<b>% Departamentos mas afectados por el dengue 2019 <b>', hole = .5,
             color_discrete_sequence=px.colors.qualitative.G10)

# poner detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    annotations = [dict(text = str(cnt_ed), x=0.5, y = 0.5, font_size = 40, showarrow = False )])
fig.show()

In [67]:
d3= dengue19.groupby(['departamento_nombre','provincia_nombre'])[['cantidad_casos']].sum().reset_index()
d3= d3.sort_values('cantidad_casos', ascending=False)
d3

departamento_nombre provincia_nombre  cantidad_casos
49                        ledesma            jujuy             536
39        grl. jose de san martin            salta             403
46                     la capital         santa fe             367
60                           oran            salta             295
41                         iguazu         misiones             253
..                            ...              ...             ...
44                    jose c. paz     buenos aires               1
45                     la caldera            salta               1
50  libertador general san martin            chaco               1
54          mayor luis j. fontana            chaco               1
83                         zarate     buenos aires               1

[84 rows x 3 columns]

En la anteiror grafica y tabla, se puede observar que para el año 2019 el departamento mas afectado fue el departamento ledesma de la provincia jujuy con un total de 536 casos, asi mismo se puede observar que la capital con mas casos es la capital de la provincia santa fe con un total de 367 casos.

**Departamento mas afectado en el 2020**

In [68]:
df4= df2.sort_values('2020', ascending=False)
df4=df4.head(5)


In [69]:
cnt_ed = df4['2020'].sum()

# hacer la gráfica
fig = px.pie(df4, values = '2020', names ='departamento_nombre',
             title= '<b>% Departamentos mas afectados por el dengue 2020 <b>', hole = .5,
             color_discrete_sequence=px.colors.qualitative.G10)

# poner detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    annotations = [dict(text = str(cnt_ed), x=0.5, y = 0.5, font_size = 40, showarrow = False )])
fig.show()

In [70]:
d2= dengue20.groupby(['departamento_nombre','provincia_nombre'])[['cantidad_casos']].sum().reset_index()
d2= d2.sort_values('cantidad_casos', ascending=False)
d2

departamento_nombre provincia_nombre  cantidad_casos
65                   capital          tucumán          6089.0
86                  comuna 1             caba          3619.0
167  grl. josé de san martín            salta          3581.0
309                san pedro            jujuy          2928.0
57                   capital          córdoba          2181.0
..                       ...              ...             ...
245               pehuenches          neuquén             1.0
118                  el alto        catamarca             1.0
122                 ensenada     buenos aires             1.0
123                escalante           chubut             1.0
173                güer aike       santa cruz             1.0

[347 rows x 3 columns]

De la anterior grafica y tabla obtenida, se puede observar que el departamento con mas casos registrados efectivamente es la capital, la cual pertenece a la provincia tucuman obteniendo un total de 6089 casos.

**Departamento mas afectado en el 2021**

In [71]:
d1= zika_21.groupby(['departamento_nombre','provincia_nombre'])[['cantidad_casos']].sum().reset_index()
d1= d1.sort_values('cantidad_casos', ascending=False)
d1

departamento_nombre provincia_nombre  cantidad_casos
48                     orán            salta            1064
10                  capital         la rioja             997
34  grl. josé de san martín            salta             546
49                   patiño          formosa             303
43                  ledesma            jujuy             218
..                      ...              ...             ...
1           almirante brown     buenos aires               1
37                ituzaingó       corrientes               1
38               la caldera            salta               1
40                   la paz        catamarca               1
33       general san martín     buenos aires               1

[66 rows x 3 columns]

In [72]:
df5= df2.sort_values('2021', ascending=False)
df5=df5.head(5)


En el caso del año 2021 se puede observar que el departamento con mas casos registrados fue el departamento de oran, el cual pertenece a la provincia de salta, con un total de 1064 casos y la capital con mas casos pertene a la provincia la rioja con un total de 997 casos

In [73]:
cnt_ed = df5['2021'].sum()

# hacer la gráfica
fig = px.pie(df5, values = '2021', names ='departamento_nombre',
             title= '<b>% Departamentos mas afectados por el dengue 2019 <b>', hole = .5,
             color_discrete_sequence=px.colors.qualitative.G10)

# poner detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    annotations = [dict(text = str(cnt_ed), x=0.5, y = 0.5, font_size = 40, showarrow = False )])
fig.show()

Con los datos obtenidos anteriormente se puede concluir que el departamento con mas casos registrados es el conjuto de las capitales de argentina, y que en cada año varia el departamento mas afectado, pero se puede concluir que los departamentos mas afectados son ledesma, oran y las capitales.

##**Conclusiones**

1. segun la noticia "Tras el récord histórico de dengue, preocupa su regreso en verano cuando todavía no esté la vacuna contra el coronavirusen" publicada en mundo sano notifica que en los años 2019-2020 se ha atraveso el brote de dengue de mayor magnitud que se haya registrado hasta el momento en Argentina con 58.889, superando casi por un 40,5% los casos acomulados, si se le mide en casos por año calendario, en las primeras 30 semanas de 2020 se registraron 58.319 casos mientras que en 2016 se registraron 40.810 casos para el mismo período, teniendo en cuenta que el 2019 y el 2020 fueron los años con mas casos segun mundo sano, tambien confirmaron que el rango edad mas afectado es entre los 20 y 34 años presentando la tasa de incidencia mas elevada, lo cual se puede verificar con el analisis presentado en este caso de estudio, donde se concluye que las personas mas afectadas por el virus entan entre los 24 y 35 años de edad.

2. El ministerio de salud de argentina  explicó que las zonas de mayor incidencia son el centro, nordeste y noroeste del país, debido a sus altas y calidas  temperaturas como es el caso de la provincia  salta la cual esta localizada en una zona tropical, posee clima calido aunque con variaciones bastante marcadas en distintas regiones y esto se debe a su relieve, por otro lado se encuentra la provincia la rioja  que se caracteriza en general por tener altas temperatura, vientos calidos y secos, en conclusion la temperatura de estas regiones es una causa de incidencia de casos de dengue.



##**Referencia**



*   Vigilancia de las enfermedades por virus del Dengue y Zika. (s. f.). Vigilancia de las enfermedades por virus del Dengue y Zika. http://datos.salud.gob.ar/dataset/vigilancia-de-dengue-y-zika

*   Sano, M. (2020, 26 agosto). Tras el récord histórico de dengue, preocupa su regreso en verano cuando todavía no esté la vacuna contra el coronavirus. Mundo Sano. Recuperado 17 de enero de 2022, de https://www.mundosano.org/es/tras-el-record-historico-de-dengue-preocupa-su-regreso-en-verano-cuando-todavia-no-este-la-vacuna-contra-el-coronaviru

*   Aznarez, C. (2020, 15 abril). Alertan por aumento de casos de dengue en Argentina. telesurtv.net. https://www.telesurtv.net/news/argentina-alertan-aumento-casos-dengue-20200415-0035.html

*   Soubie, D. E. (2020, 28 mayo). Clima de Salta. Todo Argentina. https://www.todo-argentina.net/geografia/provincias/salta/clima.html





